libr/core/carg.c

```diff
@@ -0,0 +1,248 @@
+/* radare - LGPL - Copyright 2009-2018 - pancake */
+
+#include <r_core.h>
+
+#define MAXSTRLEN 50
+
+static void set_fcn_args_info(RAnalFuncArg *arg, RAnal *anal, const char *fcn_name, const char *cc, int arg_num) {
+	if (!fcn_name || !arg || !anal) {
+		return;
+	}
+	arg->name = r_anal_type_func_args_name (anal, fcn_name, arg_num);
+	arg->orig_c_type = r_anal_type_func_args_type (anal, fcn_name, arg_num);
+	if (!strncmp ("const ", arg->orig_c_type, 6)) {
+		arg->c_type = arg->orig_c_type + 6;
+	} else {
+		arg->c_type = arg->orig_c_type;
+	}
+	const char *query = sdb_fmt (-1, "type.%s", arg->c_type);
+	arg->fmt = sdb_const_get (anal->sdb_types, query, 0);
+	const char *t_query = sdb_fmt (-1, "type.%s.size", arg->c_type);
+	arg->size = sdb_num_get (anal->sdb_types, t_query, 0) / 8;
+	arg->cc_source = r_anal_cc_arg (anal, cc, arg_num + 1);
+}
+
+static char *resolve_fcn_name(RAnal *anal, const char *func_name) {
+	const char *str = func_name;
+	const char *name = func_name;
+	if (r_anal_type_func_exist (anal, func_name)) {
+		return strdup (func_name);
+	}
+	while ((str = strchr (str, '.'))) {
+		name = str + 1;
+		str++;
+	}
+	if (r_anal_type_func_exist (anal, name)) {
+		return strdup (name);
+	}
+	return r_anal_type_func_guess (anal, (char*)func_name);
+}
+
+static ut64 get_buf_val(ut8 *buf, int endian, int width) {
+	return (width == 8)? r_read_ble64 (buf, endian) : (ut64) r_read_ble32 (buf,endian);
+}
+
+static void print_format_values(RCore *core, const char *fmt, bool onstack, ut64 src) {
```

Comment:
I think this function can be generalized and moved somewhere in libr/util or so

Reply:
There's already a function to print according to given format ('pf' command) , maybe we can generalize that , it's just that i didin't like the output of that , so implemented something minimalistic , so that i could tweak to my needs (like printing '-->' for pointers and strings)